src/common/otp.h

```diff
@@ -1,31 +1,32 @@
 // otp.h - OTP memory mapping
-#ifndef _OTP_H
-#define _OTP_H
+#pragma once
 
-// board revision (three bytes maj.min.sub)
-#define OTP_BOARD_REVISION_ADDR 0x1FFF7800
-#define OTP_BOARD_REVISION_SIZE 3
+#include <inttypes.h>
 
-// timestamp (seconds since 1970, little-endian)
-#define OTP_BOARD_TIME_STAMP_ADDR 0x1FFF7804
-#define OTP_BOARD_TIME_STAMP_SIZE 4
+enum {
+    // board revision (three bytes maj.min.sub)
+    OTP_BOARD_REVISION_ADDR = 0x1FFF7800,
+    OTP_BOARD_REVISION_SIZE = 3,
 
-// serial number without first four characters "CZPX" (total 15 chars, zero terminated)
-#define OTP_SERIAL_NUMBER_ADDR 0x1FFF7808
-#define OTP_SERIAL_NUMBER_SIZE 16
+    // timestamp (seconds since 1970, little-endian)
```

Comment:
this looks weird being trapped in an enum ... each element looks different, they cannot be ordered meaningfully :( ...

Reply:
Looks but proper solution is constexpr which is not applicable here. So this is second best.